src/sentry/interfaces/security.py

```diff
@@ -354,9 +354,22 @@ def from_raw(cls, raw):
 
         # For CSP, the values we want are nested under the 'csp-report' key.
         raw = raw['csp-report']
+
+        # For Firefox, parse effective-directive from violated-directive but
+        # only when missing since it increases the frequency of default-src
+        #
+        # refs: https://bugzil.la/1192684#c8
+        if 'effective-directive' not in raw:
+            try:
+                raw['effective-directive'] = raw['violated-directive'].split(None, 1)[0]
+            except (KeyError, IndexError):
+                pass
```

Comment:
Instead of removing `effective-directive` from the interface schema, could we munge this value before validation?

Reply:
Yeah! That's nicer I can leave the schema the same, revert the changes to event_manager, and handle everything in a `try ... except` like Matt originally suggested.